src/actions/sso/credentialRequest.ts

```diff
@@ -25,6 +26,9 @@ export const consumeCredentialRequest = (
       params: {
         interactionId: interaction.id,
         interactionSummary: interaction.getSummary(),
+        availableCreds: await (interaction.flow as CredentialRequestFlow).getAvailableCredentials(
```

Comment:
probably the messiest part, this casting should be removed and the getAvailableCredentials should be moved to InteractionManager (or even storage lib) as it's just a query and filter

Reply:
it was for the UI, but I dont think the logic/state of the interaction should be determined or affected by UI concerns, especially given that this code is moving into the SDK/a more generic application environment

Reply:
It's kind of tricky, there is a distinction I think between the interaction state (the token contents themselves) and the potential response (availableCredentials). both are required for the UI, but only one is part of the interaction, reasoning here being that before any credentials have been sent back, they are not actually part of the interaction and do not belong in it's state